app/models/cloud_volume.rb

```diff
@@ -9,15 +9,13 @@ class CloudVolume < ApplicationRecord
   belongs_to :availability_zone
   belongs_to :cloud_tenant
   belongs_to :base_snapshot, :class_name => 'CloudVolumeSnapshot', :foreign_key => :cloud_volume_snapshot_id
-  has_many   :cloud_volume_snapshots
-  has_many   :attachments, :class_name => 'Disk', :as => :backing
+  has_many :cloud_volume_snapshots
+  has_many :attachments, :class_name => 'Disk', :as => :backing
+  has_many :hardwares, :through => :attachments
```

Comment:
I would have to do:
private :hardwares, :hardwares=

below, should I do that? Not sure I've seen that pattern around. But I guess it would make sense, though lot of other places are like this

Best if rails could support multiple through on one line, so I don't have to expose all the middle layers. :-)